pkg/controller/nodeipam/ipam/multi_subnetworks_handler_test.go

```diff
@@ -0,0 +1,272 @@
+			Name: "default",
+		},
+		// Status is initially left empty
+	}
+	ntClient := ntfakeclient.NewSimpleClientset(emptyNodeTopologyCR)
+
+	tests := []struct {
+		name            string
+		node            *v1.Node
+		existingSubnets []string
+		wantSubnets     []string
+	}{
+		{
+			name: "node's subnet already exists in the cr",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						testNodePoolSubnetLabelPrefix: "subnet-def",
+						"another-label":               "value",
+					},
+				},
+			},
+			existingSubnets: []string{"subnet-def"},
+			wantSubnets:     []string{"subnet-def"},
+		},
+		{
+			name: "node has a new subnet",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						testNodePoolSubnetLabelPrefix: "new-subnet",
+					},
+				},
+			},
+			existingSubnets: []string{"subnet-def"},
+			wantSubnets:     []string{"subnet-def", "new-subnet"},
+		},
+		{
+			name: "node has a subnet and cr's subnets is empty",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						testNodePoolSubnetLabelPrefix: "subnet-def",
+					},
+				},
+			},
+			existingSubnets: []string{},
+			wantSubnets:     []string{"subnet-def"},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// populate the def subnet URL
+			// pass the fake nodetopology client
+			// create a fake default cr (outside this loop. at the begining of the test)
+			// populate the existingSubnets in the cr
+			// call the ca.updateNodeTopology and read the subnets
+			// verify the subnets with wantSubnets
+			addSubnetsToCR(tc.existingSubnets, ntClient)
+			ca := &cloudCIDRAllocator{
+				cloud:              fakeGCE,
+				nodeTopologyClient: ntClient,
+			}
+			err := ca.updateNodeTopology(tc.node)
+			if err != nil {
+				t.Errorf("ca.updateNodeTopology() returned error: %v", err)
+			}
+			if !verifySubnetsInCR(t, tc.wantSubnets, ntClient) {
+				t.Errorf("ca.updateNodeTopology() returned incorrect subnets")
+			}
+		})
+	}
+}
+
+func addSubnetsToCR(subnets []string, client ntclient.Interface) {
+	ctx := context.Background()
+
+	cr := &ntv1.NodeTopology{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default",
+		},
+		// Status is initially empty
+	}
+	for _, subnet := range subnets {
+		cr.Status.Subnets = append(cr.Status.Subnets, ntv1.SubnetConfig{
+			Name: subnet,
+			// We are ignoring the path field here on purpose.
+			// It is tested separately on it's own.
+		})
+	}
+	client.NetworkingV1().NodeTopologies().UpdateStatus(ctx, cr, metav1.UpdateOptions{})
+}
+
+func verifySubnetsInCR(t *testing.T, subnets []string, client ntclient.Interface) bool {
+	ctx := context.Background()
+	hm := make(map[string]bool)
+	for _, subnet := range subnets {
+		hm[subnet] = true
+	}
+
+	cr, _ := client.NetworkingV1().NodeTopologies().Get(ctx, "default", metav1.GetOptions{})
+
+	for _, subnetConfig := range cr.Status.Subnets {
```

Comment:
Add a size comparison to make sure subnets list is not a superset of CR's subnets?
Currently just verifies "CR's subnets are all in the specified subnets list".